datalad/dataset/gitrepo.py

```diff
@@ -53,6 +63,48 @@
 
 lgr = logging.getLogger('datalad.dataset.gitrepo')
 
+preferred_encoding = getpreferredencoding(do_setlocale=False)
+
+
+@contextmanager
+def lock_if_required(read_only,
+                     write_lock):
+    if not read_only:
+        write_lock.acquire()
+    try:
+        yield write_lock
+    finally:
+        if not read_only:
+            write_lock.release()
```

Comment:
What about the following change, and moving this into `datalad.utils`?
```suggestion
@contextmanager
def lock_if_required(lock_required, lock):
    """Acquire/ and release the provided lock if indicated by a flag"""
    if lock_required:
        lock.acquire()
    try:
        yield lock
    finally:
        if lock_required:
            lock.release()
```

Reply:
Very good

Reply:
Will move it